coderd/workspacesearch.go

```diff
@@ -0,0 +1,80 @@
+package coderd
+
+import (
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+// WorkspaceSearchQuery takes a query string and breaks it into its queryparams
+// as a set of key=value.
+func WorkspaceSearchQuery(query string) (map[string]string, error) {
```

Comment:
It seems odd that this would be exported from the `coderd` package. I'd personally prefer this be internal and tested from the endpoint itself to prevent any implementation mismatches. I'm open to thoughts here!

Reply:
nit: I noticed `AuthorizeFilter` is exposed too, and it'd be nice to make that internal as part of this PR.

Reply:
Similar comment for the `Authorize` function. It's exposed on the primary `API` struct, but it doesn't need to be from what it seems.

Reply:
I can unexport it, but I really much rather just test this pure function. Having to interpret what the results of this is based on the api feedback of the endpoint sound challenging. Testing this function as is is trivial.